pkg/api/validation/validation_test.go

```diff
@@ -1866,6 +1805,160 @@ func TestValidatePod(t *testing.T) {
 				DNSPolicy:     api.DNSClusterFirst,
 			},
 		},
+		{ // Serialized pod affinity in affinity requirements in annotations.
```

Comment:
Please break out these tests separately as their own test case, otherwise we are conflating the unit test cases.